pilot/pkg/xds/endpoint_builder.go

```diff
@@ -551,7 +551,7 @@ func trafficPolicyTLSModeForPort(tp *networkingapi.TrafficPolicy, port int) *net
 	}
 	// if there is a port-level setting matching this cluster
 	for _, portSettings := range tp.GetPortLevelSettings() {
-		if int(portSettings.Port.Number) == port && portSettings.Tls != nil {
+		if int(portSettings.GetPort().GetNumber()) == port && portSettings.Tls != nil {
```

Comment:
https://github.com/istio/istio/blob/066f4b429663f2cd7afce744b2a8bdde7d1103a9/pkg/config/validation/validation.go#L1525

did you disable the validating webhook?

Reply:
😮 yes 
Due to the misconfiguration of Istio installation parameters `values.global.configValidation: false`.

The chain reaction causes the creation of incorrect `DestinationRule` resources, causing Pilot to crash.

Reply:
but, I think it is necessary here. 
A mis-usage can crash Istiod. ⚠️